multipart/src/main.rs

```diff
@@ -80,18 +58,20 @@ fn index() -> HttpResponse {
 
 fn main() -> std::io::Result<()> {
     std::env::set_var("RUST_LOG", "actix_server=info,actix_web=info");
-    env_logger::init();
-
+    std::fs::create_dir_all("./tmp").unwrap();
+    let config = envy::from_env::<Config>().unwrap();
+    let p =  config.port;
+    let port = concat_string!("0.0.0.0:", p);
     HttpServer::new(|| {
-        App::new()
-            .data(Cell::new(0usize))
-            .wrap(middleware::Logger::default())
-            .service(
-                web::resource("/")
-                    .route(web::get().to(index))
-                    .route(web::post().to_async(upload)),
-            )
+    App::new()
+    .data(Cell::new(0usize))
```

Comment:
anyone have an explanation for this line?

Reply:
https://github.com/actix/examples/blob/master/multipart/src/main.rs#L52

Reply:
Oh, wow, that's a little unintuitive, no? `.data()` is a builder for AppState?

Reply:
![Screenshot from 2019-12-04 12-11-59](https://user-images.githubusercontent.com/320177/70168892-5f1b9900-168f-11ea-8e6e-889b54fa84e9.png)

I wouldn't say that, I think it's simply derelict.